packages/dev-server/src/MetroDevServer.ts

```diff
@@ -162,12 +168,24 @@ export async function bundleAsync(
     return await Promise.all(
       bundles.map(async (bundle: BundleOptions) => {
         const bundleOutput = await buildAsync(bundle);
-        const shouldBuildHermesBundle = await shouldBuildHermesBundleAsync(
+        const isHermesManaged = isEnableHermesManaged(expoConfig, bundle.platform);
+
+        const maybeInconsistentEngine = await maybeInconsistentEngineAsync(
           projectRoot,
-          bundle.platform
+          bundle.platform,
+          isHermesManaged
         );
+        if (maybeInconsistentEngine) {
+          throw new Error(
+            `Inconsistent Hermes settings between app config and ${bundle.platform} native project.\n` +
+              `Enabling Hermes in app.json or app.config.js: ${isHermesManaged.toString()}\n` +
```

Comment:
let's use [getConfigFilePaths](https://github.com/expo/expo-cli/blob/a02c8997980d59007478d2c29a792276962147c3/packages/config/src/Config.ts#L257-L272) to output the config file used. if dynamic config path is available, we specify that, otherwise we use the static path